lib/java/com/google/android/material/imageview/ShapeableImageView.java

```diff
@@ -219,13 +219,17 @@ public void setContentPadding(
     startContentPadding = UNDEFINED_PADDING;
     endContentPadding = UNDEFINED_PADDING;
 
-    // Super padding is equal to background padding + content padding. Adjust the content padding
-    //  portion of the super padding here:
-    super.setPadding(
-        super.getPaddingLeft() - leftContentPadding + left,
-        super.getPaddingTop() - topContentPadding + top,
-        super.getPaddingRight() - rightContentPadding + right,
-        super.getPaddingBottom() - bottomContentPadding + bottom);
+    // If onMeasure hasn't adjusted padding yet, wait for it to be set to avoid double-applying the
+    //  content padding in onMeasure:
+    if (hasAdjustedPaddingAfterLayoutDirectionResolved) {
+      // Super padding is equal to background padding + content padding. Adjust the content padding
+      //  portion of the super padding here:
+      super.setPadding(
```

Comment:
It looks like there might be a bug if content padding are set twice, and the first time hasAdjustedPaddingAfterLayoutDirectionResolved is false and the second time it's true. 

At the second call, super's paddings are not actually updated with content paddings yet, so subtracting the content paddings will causes extra paddings being subtracted. Can you also add test cases for this situation after you fix it?

Reply:
At your second call to `setContentPadding`, super's paddings **have** been updated, because `setPadding` is called in `onMeasure` immediately after `hasAdjust...` was set, and `setPadding` then called `super.setPadding` with the added differences.

In any case I'll look to add tests that check this.